lang/step.md

```diff
@@ -359,19 +359,30 @@ impl<M: Memory> Machine<M> {
         };
         let mut locals: Map<LocalName, Place<M>> = default();
 
+        // If the callee might return, then `ret` and `next_block` should be `Some`.
+        // And if it never returns, they are required to be `None`.
+        let fn_might_return = func.ret.is_some();
+        if fn_might_return != ret.is_some() || fn_might_return != next_block.is_some() {
+            throw_ub!("Contradicting information on whether function might return");
+        }
```

Comment:
Instead of using `match` or `if let` to access `ret` and `next_block` I prefer to check `fn_might_return` and then unwrap.
This makes it clear that there are only two options:
`func.ret, ret, next_block` are all `Some` or `None`.

Do you agree with this? This would mean that `-> ()` functions do return like normal functions.
And only `-> !`, `start` and `Exit` have `None`, I think.

Reply:
This looks mighty complicated and I don't understand the motivation for doing this. I'd not couple things together like that. Instead just cause UB when information is missing that you absolutely need at that moment.